File: src/test/java/com/qomolangma/api/PingUseCase.java
package com.qomolangma.api;

import com.qomolangma.frameworks.application.core.UseCase;
import com.qomolangma.frameworks.payload.core.RequestPayload;
import com.qomolangma.frameworks.payload.core.ResponsePayload;

import javax.annotation.Resource;

@UseCase
public class PingUseCase {
    private @Resource Ping ping;

    public ResponsePayload execute(RequestPayload payload) {
        return ping.pong(payload);
    }
}

File: src/test/kotlin/com/qomolangma/api/PingUseCase.kt
package com.qomolangma.api

import com.qomolangma.frameworks.application.core.UseCase
import com.qomolangma.frameworks.payload.core.Payload
import javax.annotation.Resource

@UseCase
class PingUseCase {
    @Resource
    private val ping: Ping? = null

    fun execute(payload: Payload): Payload {
        return ping!!.pong(payload)
    }
}
